devp2p/discovery.py

```diff
@@ -218,7 +222,7 @@ def __init__(self, app, transport):
 
     def get_node(self, nodeid, address=None):
         "return node or create new, update address if supplied"
-        assert isinstance(nodeid, str)
+        assert isinstance(nodeid, bytes)
```

Comment:
This statement isn't exactly equivalent in 2/3 since it will return `True` for both byte and text strings in python 2 but only true for byte strings in python3 .